openedx/frontend-app-course-authoring
# feat: Add "Recently Modified" library section [FC-0059]

## Description

This PR adds the "Recently Modified" section in the Library Home Page along with the "View All" button that appears in other sections if there is more content than what is previewed.

**Recently Modified Section**
This section shows the most recently modified/created **4 or less** library components. If there are no components yet in the library the section will not appear on the page. This section will always show those components regardless of whatever search terms or filters have been applied.

<img width="1355" alt="Screenshot 2024-07-12 at 1 34 30 PM" src="https://github.com/user-attachments/assets/06623560-3f14-46a2-a944-0f0c09908803">


**View All button**
The "View All" button appears on all sections in the library page (excluding the "Recently Modified" section) when the number of content in the section exceeds what is being previewed (currently that limit is **4**). Clicking on the "View All" button should navigate to the sections Tab to view all the content, while maintaining any search terms/filters/sorting applied.

<img width="1345" alt="Screenshot 2024-07-12 at 1 35 37 PM" src="https://github.com/user-attachments/assets/e5846cc5-d513-48e7-a50d-a880aa01e798">

## Supporting information

Related Tickets:
- https://github.com/openedx/frontend-app-course-authoring/issues/1039
- Depends on https://github.com/openedx/edx-platform/pull/35103

## Testing instructions

1. Make sure you have [edx-platform](https://github.com/openedx/edx-platform/pull/35103) running locally
1. Run this branch locally
1. Make sure you have a few libraries created (you can use the soon-to-be-deprecated [library-authoring mfe](https://github.com/openedx/frontend-app-library-authoring)):
    1. A library that has 0 components
    1. A library that has more than 4 components
1. After creating/updating the libraries, make sure to run `./manage.py cms reindex_studio --experimental` in the cms container to update the search indexes with latest data
1. In the course-authoring mfe, navigate to the libraries tab
1. Click on the library that has 0 components:
    1.  Confirm that the it shows the empty state, and no "Recently Modified" section
1. Click on the library that has more than 4 components:
    1. Confirm that the "Recently Modified" section is shown in the home tab
    1. Confirm that the components show are in fact the most recently modified
    1. Confirm that performing a search/filtering/sorting has no impact on that section
    1. Confirm that the "View All" button is shown in the components section when there are more than 4 results
    1. Confirm that clicking on the "View All" button navigates to the "Components" tab, while maintaining any sorting/filtering/search terms applied
    1. Perform a filter or search to get 4 or less results, and confirm that the "View All" button is no longer displayed and that the number inside the components is updated to reflect how many components match
1. Confirm that the modifying a component in the library (using the library-authoring mfe or shell) and updating the search index (see cms command above) will include that component at the beginning of the "Recently Modified" section.

---
Private-ref: [FAL-3759](https://tasks.opencraft.com/browse/FAL-3759)
